### PR TITLE
[WFLY-14123] Add OSGI Headers for jboss-client.jar

### PR DESCRIPTION
**Pull request by moulalis**
Issue: https://issues.redhat.com/browse/WFLY-14123

### CODE REVIEW

File: client/shade/pom.xml [hunk 1]
```diff
@@ -279,6 +279,13 @@
                                 <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
                                     <manifestEntries>
                                         <Multi-Release>true</Multi-Release>
+                                        <Bundle-ManifestVersion>2</Bundle-ManifestVersion>
+                                        <Bundle-SymbolicName>org.jboss.client</Bundle-SymbolicName>
+                                        <Bundle-Version>1.0</Bundle-Version>
+                                        <Bundle-Name>Test</Bundle-Name>
```
Comment (yersan) [inline client/shade/pom.xml:285]: The Bundle-Name should be something more descriptive than "Test".

A good candidate could be something derived from client/shade/pom.xml#name, although without "WildFly: "

**Review by yersan — CHANGES_REQUESTED** [review]

**Comment by moulalis** [comment]
@yersan I made the changes to name, please review.

File: client/shade/pom.xml (hunk 1 above)
Comment (moulalis) [reply client/shade/pom.xml]: @yersan I made the changes to the bundle name, please review.

File: client/shade/pom.xml [hunk 2]
```diff
@@ -279,6 +279,13 @@
                                 <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
                                     <manifestEntries>
                                         <Multi-Release>true</Multi-Release>
+                                        <Bundle-ManifestVersion>2</Bundle-ManifestVersion>
+                                        <Bundle-SymbolicName>org.jboss.client</Bundle-SymbolicName>
+                                        <Bundle-Version>1.0</Bundle-Version>
+                                        <Bundle-Name>OSGI Header</Bundle-Name>
```
Comment (yersan) [inline client/shade/pom.xml:285]: Hello @moulalis , it should be a name describing what is the bundle content.  `OSGI Header` still doesn't say too much.

**Review by yersan — CHANGES_REQUESTED** [review]

File: client/shade/pom.xml (hunk 2 above)
Comment (moulalis) [reply client/shade/pom.xml]: Hi @yersan,
I've changed it to 'JBoss Client library', if not suitable please suggest.
Thanks.

File: client/shade/pom.xml [hunk 3]
```diff
@@ -279,6 +279,13 @@
                                 <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
                                     <manifestEntries>
                                         <Multi-Release>true</Multi-Release>
+                                        <Bundle-ManifestVersion>2</Bundle-ManifestVersion>
+                                        <Bundle-SymbolicName>org.jboss.client</Bundle-SymbolicName>
+                                        <Bundle-Version>1.0</Bundle-Version>
+                                        <Bundle-Name>JBoss EAP client library</Bundle-Name>
```
Comment (yersan) [inline client/shade/pom.xml:285]: Well, this is WildFly and not EAP, so JBoss EAP I guess is not suitable here. I suggested in a previous comment to look for a name from client/shade/pom.xml#name, although without "WildFly: ".

What about "Jakarta Enterprise Beans and Jakarta Messaging client library"? Would you be fine with using it?

**Review by yersan — CHANGES_REQUESTED** [review]

File: client/shade/pom.xml (hunk 3 above)
Comment (moulalis) [reply client/shade/pom.xml:285]: Hi @yersan,
It was my bad, I was suppose to add 'Wildfly client library' and 'JBoss EAP client library' was for the JBEAP issue.
"Jakarta Enterprise Beans and Jakarta Messaging client library"? Would be good to add.
Thanks.

File: client/shade/pom.xml [hunk 4]
```diff
@@ -279,6 +279,13 @@
                                 <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
                                     <manifestEntries>
                                         <Multi-Release>true</Multi-Release>
+                                        <Bundle-ManifestVersion>2</Bundle-ManifestVersion>
+                                        <Bundle-SymbolicName>org.jboss.client</Bundle-SymbolicName>
+                                        <Bundle-Version>1.0</Bundle-Version>
+                                        <Bundle-Name>Jakarta Enterprise Beans and Jakarta Messaging client library</Bundle-Name>
+                                        <ExtensionFragment-Host>org.openjdk.jmc.rjmx</ExtensionFragment-Host>
```
Comment (yersan) [inline client/shade/pom.xml:286]: Ups, sorry @moulalis for all the back and forth on the review, could you please replace `ExtensionFragment-Host` by `Fragment-Host`, I've just realized `ExtensionFragment-Host` is not a valid OSGI bundle property. It seems the Jira description itself is badly formatted and hence all the confusion.

**Review by yersan — CHANGES_REQUESTED** [review]

File: client/shade/pom.xml (hunk 4 above)
Comment (moulalis) [reply client/shade/pom.xml]: It's done, Thanks.

**Review by yersan — APPROVED** [review]

File: client/shade/pom.xml (hunk 4 above)
Comment (yersan) [reply client/shade/pom.xml]: thank-you!

**Pull request merged**